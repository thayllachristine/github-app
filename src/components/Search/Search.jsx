import React from 'react'
import styled from 'styled-components'

import Input from './Input'
import Button from './Button'

export const StylizedSearch = styled.form`
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
`

const Search = props => {
  const { children, onSubmit } = props
  return (
    <StylizedSearch onSubmit={onSubmit}>
      {children}
    </StylizedSearch>
  )
}

Search.Input = Input
Search.Button = Button

Search.displayName = 'Search'

export default Search
