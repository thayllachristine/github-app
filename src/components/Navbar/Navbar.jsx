import React, { useState } from 'react'
import { Link, withRouter } from 'react-router-dom'
import styled from 'styled-components'
import Octions, { MarkGithub } from '@primer/octicons-react'

import Search from '../Search'
import getUser from './../../resources/getUser'
import { setLS } from './../../services/localStorage'

const StylizedNavbar = styled.nav`
  background-color: #24292e;
  display: flex;
  justify-content: space-between;
  padding: 0 15px;
`

const LinkToHome = styled(Link)`
  align-items: center;
  display: flex;
  padding: 15px;
  text-decoration: none;
  svg {
    margin-right: 15px;
  }
  p, svg { 
    color: white;
  }
`

const Navbar = withRouter(({ history }) => {
  // eslint-disable-next-line no-unused-vars
  const [user, setUser] = useState(null)
  const userRef = React.createRef()

  const handleSubmit = e => {
    e.preventDefault()
    setUser(userRef.current.value)
    let userValue = userRef.current.value
    getUser(userValue)
      .then(r => {
        setLS('user', r.data.login)
        history.push(`/${userValue}/repositories`)
      })
      .catch(e => {
        history.push('/')
      })
  }

  return (
    <StylizedNavbar>
      <LinkToHome to='/'>
        <Octions icon={MarkGithub} size='medium' ariaLabel='GitHub Mark' />
        <p>GitHub Search</p>
      </LinkToHome>
      <Search onSubmit={handleSubmit}>
        <Search.Input insideNavbar ref={userRef} />
        <Search.Button insideNavbar />
      </Search>
    </StylizedNavbar >
  )
})

export default Navbar