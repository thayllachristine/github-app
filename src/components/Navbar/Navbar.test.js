import React from 'react'
import { shallow } from 'enzyme'
import Navbar from './Navbar'
import Search from './../Search'

describe('<Navbar />', () => {
  it('should match default snapshot', () => {
    const component = shallow(<Navbar />)
    expect(component).toMatchSnapshot()
  })

  it('should have a state', () => {
    const component = shallow(<Navbar />)
    expect(component.find('user')).toHaveLength(0)
  })

  it('should have a Search component', () => {
    const component = shallow(<Navbar />)
    expect(component.find('Search')).toBeTruthy()
  })
})