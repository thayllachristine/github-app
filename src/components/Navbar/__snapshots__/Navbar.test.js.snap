// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match default snapshot 1`] = `ShallowWrapper {}`;
